src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -82,14 +89,7 @@ public void Authenticate(IConnection connection, ConnectionDescription descripti
                     {
                         throw CreateException(connection, ex);
                     }
-
-                    currentStep = Transition(conversation, currentStep, result);
-                    if (currentStep == null)
-                    {
-                        return;
-                    }
-
-                    command = CreateContinueCommand(currentStep, result);
+                    currentStep = Transition(conversation, currentStep, result , out command);
```

Comment:
You misunderstood where I was suggesting blank lines:

    public void Authenticate(IConnection connection, ConnectionDescription description, CancellationToken cancellationToken)
    {
        Ensure.IsNotNull(connection, nameof(connection));
        Ensure.IsNotNull(description, nameof(description));

        using (var conversation = new SaslConversation(description.ConnectionId))
        {
            var currentStep = _mechanism.Initialize( connection, conversation, description);
            var command = CreateStartCommand(currentStep);

            var speculativeAuthenticateResult = description.IsMasterResult.SpeculativeAuthenticate;
            if (speculativeAuthenticateResult != null)
            {
                currentStep = Transition(conversation, currentStep, speculativeAuthenticateResult, out command);
            }

            while (currentStep != null)
            {
                BsonDocument result;
                try
                {
                    var protocol = CreateCommandProtocol(command);
                    result = protocol.Execute(connection, cancellationToken);
                }
                catch (MongoCommandException ex)
                {
                    throw CreateException(connection, ex);
                }

                currentStep = Transition(conversation, currentStep, result, out command);
            }
        }
    }

Reply:
Ah sorry. Used a diff, and I think I've got it this time? :)